ckanext/googleanalytics/logic.py

```diff
@@ -0,0 +1,77 @@
+import hashlib
+import json
+import logging
+import requests
+import threading
+
+import ckan.plugins.toolkit as toolkit
+import ckan.model as model
+
+from ckanext.googleanalytics.utils.ga import (
+    _init_service,
+    _get_packages_data,
+    _save_packages_data,
+    _get_urls_data,
+    _save_urls_data,
+)
+
+
+log = logging.getLogger(__name__)
+
+
+class AnalyticsPostThread(threading.Thread):
+    def __init__(self, queue):
+        threading.Thread.__init__(self)
+        self.queue = queue
+
+    def run(self):
+        while True:
+            data = self.queue.get()
+            log.debug("Sending API event to Google Analytics: GA4")
+            measurement_id = toolkit.config.get(
+                "googleanalytics.measurement_id"
+            )
+            api_secret = toolkit.config.get("googleanalytics.api_secret")
+            res = requests.post(
+                "https://www.google-analytics.com/mp/collect?measurement_id={}&api_secret={}".format(
+                    measurement_id, api_secret
+                ),
+                data=json.dumps(data),
+                timeout=10,
+            )
+            self.queue.task_done()
+
+
+def post_analytics(user):
+    from ckanext.googleanalytics.plugin import GoogleAnalyticsPlugin
+
+    path = toolkit.request.environ["PATH_INFO"]
+    path_id = path.split("/dataset/")[1].split("/")[0]
+    context = {"model": model, "session": model.Session, "user": user}
+    package = toolkit.get_action("package_show")(context, {"id": path_id})
```

Comment:
It is my understanding that `toolkit.get_action` automatically adds the model and session and user to the context.  I believe this also means you don't need to import the model. 
```suggestion
    package = toolkit.get_action("package_show")({}, {"id": path_id})
```